numpy/meson.build

```diff
@@ -311,8 +311,23 @@ else
   )
 endif
 
+pure_subdirs_with_tests = [
+  'compat',
+  'f2py',
+  'lib',
+  'ma',
+  'matrixlib',
+  'polynomial',
+  'testing',
+  'typing'
+]
+
 foreach subdir: pure_subdirs
-  install_subdir(subdir, install_dir: np_dir, install_tag: 'python-runtime')
+  install_subdir(subdir, install_dir: np_dir, install_tag: 'python-runtime', exclude_directories: ['tests'])
+endforeach
+
+foreach subdir: pure_subdirs_with_tests
+  subdir(subdir)
 endforeach
```

Comment:
@rgommers The thing we did in our meeting doesn't seem to work i.e., `install_subdir(subdir, install_dir: np_dir, install_tag: 'python-runtime', exclude_directories: ['tests'])` and `install_subdir(subdir/'tests', install_dir: np_dir, install_tag: 'tests')`. So I had to do `subdir` inside the directories having `tests` folders. The changes I did produced results i.e., `tests` are gone if no `tests` tag is present in `install-tags`.

Reply:
What goes wrong? This seems to work for me (note the difference in `install_dir` to what you wrote above):
```diff
diff --git a/numpy/meson.build b/numpy/meson.build
index 80fa720b82..6cc10f4d6d 100644
--- a/numpy/meson.build
+++ b/numpy/meson.build
@@ -312,7 +312,8 @@ else
 endif

 foreach subdir: pure_subdirs
-  install_subdir(subdir, install_dir: np_dir, install_tag: 'python-runtime')
+  install_subdir(subdir, install_dir: np_dir, install_tag: 'python-runtime', exclude_directories: ['tests'])
+  install_subdir(subdir/'tests', install_dir: np_dir/subdir, install_tag: 'tests')
 endforeach

 compilers = {
```

Reply:
I think `numpy/tests` should also be handled separately from the loop above.

Reply:
Yes indeed, `numpy/tests` needs to be separate.

Reply:
Ah, I guess that was the answer as to what goes wrong:
```
    File "/tmp/pip-build-env-gd4z7d5a/overlay/lib/python3.11/site-packages/mesonpy/__init__.py", line 809, in _manifest
      return _map_to_wheel(sources)
             ^^^^^^^^^^^^^^^^^^^^^^
    File "/tmp/pip-build-env-gd4z7d5a/overlay/lib/python3.11/site-packages/mesonpy/__init__.py", line 122, in _map_to_wheel
      assert os.path.isdir(src)
             ^^^^^^^^^^^^^^^^^^
```

Things work with `spin build`, but not with building a wheel. It looks like it needs guarding with [`fs.is_dir(subdir/'tests')`](https://mesonbuild.com/Fs-module.html#is_dir)

Reply:
Yes. Works now. :-)